src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useContext, useState, useEffect, useRef } from 'react';
+import { Todo } from '../Todo/Todo';
+import { Notification } from '../Notification/Notification';
+import { TodoContext } from '../../contexts/TodoContext';
+import { USER_ID, createTodo, deleteTodo, updateTodo } from '../../api/todos';
+import {
+  FilterContext,
+  ACTIONSINFILTERCONTEXT,
+} from '../../contexts/FilterContext';
+import { Filter } from '../Filter/Filter';
+
+export const TodoApp: React.FC = () => {
+  const { todosFromServer, refreshTodos, putErrorWarning } =
+    useContext(TodoContext)!;
+  const {
+    filteredTodos,
+    currentFilterCriteria,
+    itemsLeft,
+    totalCompletedTodos,
+  } = useContext(FilterContext)!;
```

Comment:
And here also

Reply:
Expanation about use undefined in filterContext in other parts of code, here is certain that is not null